### PR TITLE
refactor: Migrate to new file actions for 28

### PR DESCRIPTION
**Pull request by provokateurin**
The way I did it might not be the cleanest but I found it to be the least amount of work:
1. Copy the filesplugin.js from integration_mattermost
2. Apply all the changes from https://github.com/nextcloud/integration_mattermost/compare/main...nextcloud:integration_slack:main (only filesplugin.js) that are Slack specific.
3. Adjust the min version and dependencies

### CODE REVIEW

File: src/filesplugin.js [hunk 1]
```diff
@@ -34,166 +51,130 @@ function openChannelSelector(files) {
...
+		return !OCA.Slack.actionIgnoreLists.includes(view.id)
+			&& nodes.length > 0
+			&& nodes.every(({ permissions }) => (permissions & Permission.READ) !== 0)
+		// && nodes.every(({ type }) => type === FileType.File)
+		// && nodes.every(({ mime }) => mime === 'application/some+type')
+	},
+	iconSvgInline: () => SlackIcon,
+	async exec(node, view, dir) {
+		sendSelectedNodes([node])
+		return true
```
Comment (kyteinsky) [inline src/filesplugin.js:68]: some more improvements: https://github.com/nextcloud/integration_mattermost/pull/30/files

File: src/filesplugin.js [hunk 2]
```diff
@@ -263,100 +344,21 @@ OCA.Slack.SlackSendModalVue.$on('closed', () => {
...
-function errorCallback(error) {
-	console.error(error)
-	OCA.Slack.SlackSendModalVue.failure()
-	OCA.Slack.filesToSend = []
-	OCA.Slack.sentFileNames = []
-}
-
 // get Slack state
-axios.get(IS_CONNECTED_URL).then((response) => {
+const urlCheckConnection = generateUrl('/apps/integration_slac/is-connected')
```
Comment (kyteinsky) [inline src/filesplugin.js:360]: '/apps/integration_slack/is-connected'  
That one took long to pin point :)

Comment (provokateurin) [reply src/filesplugin.js:360]: oops

Comment (provokateurin) [reply src/filesplugin.js:360]: I checked the whole file and I only made this typo once.

**Comment by kyteinsky** [comment]
@provokateurin Can you adopt [this code](https://github.com/nextcloud/integration_slack/blob/main/src/filesplugin.js#L196-L356) and replace the counterparts of the mattermost code in `filesplugin.js`. This uses Promise.all to send all files at once instead of a loop and also should solve the "no-text" error from Slack thrown when a file is sent without any comment. Also, there is a model not closing on its own after the files are sent. I did not look into it yet but I think porting this back should mitigate other undiscovered bugs.

(Just a suggestion: only changing the file context menu part in `filesplugin.js` for that file should be easier with lesser modifications)

**Comment by provokateurin** [comment]
Yeah I saw that you made these improvements. I can try to put it on top, although that will then require new testing from you.

**Comment by provokateurin** [comment]
Ideally those improvements should also be made to the mattermost integration cc @julien-nc

**Comment by kyteinsky** [comment]
> Yeah I saw that you made these improvements. I can try to put it on top, although that will then require new testing from you.

I would still push on only modifying parts rather than replacing everything and then inserting the changes but YMMV. I can test it whenever you're done, or you can try setting it up for faster iterations. You will need a https NC

> Ideally those improvements should also be made to the mattermost integration cc julien-nc

Yeah I intend to port them later on, this or the next week probably.

**Comment by julien-nc** [comment]
> Ideally those improvements should also be made to the mattermost integration

It's nice indeed but if the network speed is the bottleneck, sending all files in parallel won't change much. I'm fine if Mattermost still sends them sequentially.

File: src/filesplugin.js [hunk 3]
```diff
@@ -356,7 +342,8 @@ function errorCallback(error) {
 }
 
 // get Slack state
-axios.get(IS_CONNECTED_URL).then((response) => {
+const urlCheckConnection = generateUrl(IS_CONNECTED_URL)
```
Comment (kyteinsky) [inline src/filesplugin.js:345]: `IS_CONNECTED_URL` is an already generated URL

**Comment by kyteinsky** [comment]
looks fine except one thing: folders error out and slack does not accept them so skip them in the `sendFile` function. We probably should decrease the count of successful files sent using `resolve()` and then collecting the values in `then` block.  
`if (fileType === FileType.Folder) {`

**Comment by provokateurin** [comment]
> folders error out and slack does not accept them so skip them in the sendFile function

They are already skipped before calling sendFile in line 316

> We probably should decrease the count of successful files sent using resolve() and then collecting the values in then block.

I don't understand what you mean, could you please elaborate?

**Comment by kyteinsky** [comment]
> They are already skipped before calling sendFile in line 316

didn't see the error this time but a silent ignore. This can have a separate PR but you can add it while we are at it.

```js
function noFilesError() {
	showError(t('integration_slack', 'No files to send (folders are not supported)'))
	OCA.Slack.SlackSendModalVue.failure()
}
```

Call it on line 253:
```js
	if (filesToSend.length === 0) {
		return noFilesError()
	}
```

And on line 318 after the folder check:
```js
		if (OCA.Slack.filesToSend.length === 0) {
			return noFilesError()
		}
```

> I don't understand what you mean, could you please elaborate?

Oh nevermind that now :D

**Review by kyteinsky — APPROVED** [review]
Nice work! Works as expected.

**Pull request merged**